driver-async/src/test/unit/com/mongodb/async/client/FindIterableSpecification.groovy

```diff
@@ -327,4 +327,20 @@ class FindIterableSpecification extends Specification {
         then:
         thrown(IllegalArgumentException)
     }
+
+    def 'should get and set batchSize as expected'() {
+        when:
+        def batchSize = 5
+        def mongoIterable = new FindIterableImpl(null, namespace, Document, Document, codecRegistry, readPreference,
+                readConcern, Stub(AsyncOperationExecutor), new Document(), new FindOptions())
+
+        then:
+        mongoIterable.getBatchSize() == 0
```

Comment:
This is problematic - findOptions uses int rather than Integer :/

Having a default values means that Observables based FindIterable's will always use the servers batch size - which would be a change in behaviour.  Happily, I've fixed this anomoly in #239